### PR TITLE
remove peer

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @disksing

### CODE REVIEW

File: tests/raftstore/cluster.rs [hunk 1]
```diff
@@ -140,20 +144,20 @@ impl<T: Simulator> Cluster<T> {
     }
 
     pub fn call_command(&self,
+                        store_id: u64,
```
Comment (ngaut) [inline tests/raftstore/cluster.rs]: Should we embedded store_id to RaftCmdRequest ?

Comment (siddontang) [inline tests/raftstore/cluster.rs]: @BusyJay suggests removing from it.

Comment (ngaut) [inline tests/raftstore/cluster.rs]: Every RaftCmdRequest needs to send to a store, Any benefit to  remove it?

Comment (siddontang) [inline tests/raftstore/cluster.rs]: we know which store to send outer directly. no need to embed it in request.

File: src/pd/protocol.rs [hunk 2]
```diff
@@ -106,11 +106,11 @@ impl super::PdClient for RpcClient {
     fn ask_change_peer(&self,
                        cluster_id: u64,
                        region: metapb::Region,
-                       leader: metapb::Peer)
+                       leader_store_id: u64)
```
Comment (BusyJay) [inline src/pd/protocol.rs]: I think `leader_id` is enough.

Comment (siddontang) [inline src/pd/protocol.rs]: we should emphasis store id here.

Comment (ngaut) [inline src/pd/protocol.rs]: I prefer leader_store_id. It's more clear. leader_id makes me confused.

File: src/raftstore/store/peer.rs [hunk 3]
```diff
@@ -293,7 +287,7 @@ impl Peer {
         }
 
         debug!("handle raft ready: peer {:?}, region {}",
-               self.peer,
+               self.store_id(),
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: I don't think we need to print `store_id` here.

Comment (siddontang) [inline src/raftstore/store/peer.rs]: we can see which store handles raft ready in log directly.

File: src/raftstore/store/util.rs [hunk 4]
```diff
@@ -20,33 +20,19 @@ use kvproto::raftpb::{self, ConfChangeType};
...
-    }
-
-    None
+pub fn find_peer(region: &metapb::Region, store_id: u64) -> bool {
+    region.get_store_ids().iter().any(|&id| id == store_id)
 }
 
-pub fn remove_peer(region: &mut metapb::Region, store_id: u64) -> Option<metapb::Peer> {
-    match region.get_peers()
+pub fn remove_peer(region: &mut metapb::Region, store_id: u64) -> Option<u64> {
```
Comment (BusyJay) [inline src/raftstore/store/util.rs]: I don't think we need to return store_id here.

Comment (siddontang) [inline src/raftstore/store/util.rs]: yep, we can return bool directly like HashSet now.

**Comment by ngaut** [comment]
LGTM

File: src/raftstore/store/peer.rs [hunk 5]
```diff
@@ -167,36 +163,31 @@ impl Peer {
                        from_epoch,
                        region_epoch);
                 // We receive a stale message and we can't re-create the peer with the peer id.
-                return Err(box_err!("peer {} already destroyed", peer_id));
+                return Err(box_err!("peer {} already destroyed", store_id));
```
Comment (disksing) [inline src/raftstore/store/peer.rs]: Better to log region_id here.

File: src/raftstore/store/peer.rs [hunk 6]
```diff
@@ -385,7 +379,7 @@ impl Peer {
 
         let mut cc = raftpb::ConfChange::new();
         cc.set_change_type(change_peer.get_change_type());
-        cc.set_node_id(change_peer.get_peer().get_id());
+        cc.set_node_id(change_peer.get_store_id());
```
Comment (disksing) [inline src/raftstore/store/peer.rs]: node_id -> store_id ?

Comment (siddontang) [inline src/raftstore/store/peer.rs]: no, this is defined in etcd, we should keep it.

**Comment by disksing** [comment]
LGTM.

**Pull request merged**